django_elasticsearch_dsl/fields.py

```diff
@@ -125,7 +128,7 @@ def get_value_from_instance(self, instance, field_value_to_ignore=None):
         if objs is None:
             return {}
         try:
-            is_iterable = bool(iter(objs))
+            is_iterable = bool(iter(objs)) and not isinstance(objs, dict)
```

Comment:
I think rather than changing `is_iterable`, change to `if is_iterable and not isinstance(objs, dict):` below.